### ISSUE TITLE
Decode `ContractEmitted` event

### ISSUE THREAD

**Issue by athei** [paritytech/contracts-ui#229]
When a contract execution emits a [`ContractEmitted`](https://github.com/paritytech/substrate/blob/b36030a6e8c21af9f18bb0d6a667188fac0dbc47/frame/contracts/src/lib.rs#L553-L560) event it means that it was emitted by the contract logic itself and not the `pallet-contracts`. This implies that the `data` field is opaque unless you have the metadata at hand to decode it.

The contracts-ui usually has this data and should decode the `data` field into the custom event defined by the contract before displaying it to the user.

<img width="538" alt="Screenshot 2022-03-09 at 19 37 51" src="https://user-images.githubusercontent.com/2580396/157509020-78ec244c-e606-43d8-a487-2f07ef812d8e.png">

**Comment by agryaznov** [comment]
implementation of such a decoding in `cargo-contract` can be found here:  [cargo-contract#481](https://github.com/paritytech/cargo-contract/pull/481)

**Comment by agryaznov** [comment]
Also I think it would be very helpful for contract developers if extrinsic `data`, like contract message input, is being displayed somewhere in UI, as a _copyable_ field. See [this](https://github.com/polkadot-js/apps/issues/7266) PolkadotJS apps issue for reference.

**Comment by athei** [comment]
Yeah fields of events seem to be missing in general (even from the other events).
